src/fu-util.c

```diff
@@ -1870,6 +1870,146 @@ fu_util_reinstall (FuUtilPrivate *priv, gchar **values, GError **error)
 	return fu_util_prompt_complete (priv->completion_flags, TRUE, error);
 }
 
+static gboolean
+fu_util_switch_branch_warning (FuUtilPrivate *priv, FwupdRelease *rel, GError **error)
+{
+	const gchar *desc_markup = NULL;
+	g_autofree gchar *desc_plain = NULL;
+
+	/* get formatted text */
+	desc_markup = fwupd_release_get_description (rel);
+	if (desc_markup == NULL)
+		return TRUE;
+	desc_plain = fu_util_convert_description (desc_markup, error);
+	if (desc_plain == NULL)
+		return FALSE;
+
```

Comment:
I think it would be good to have a boilerplate warning here like we do for the lvfs-testing remote

Reply:
What would it say? I don't think an alternate branch is always bad, for instance, what if System76 supported a coreboot firmware for their laptop and also a nonfree bios from AMI. They're both supported by the vendor, and both as valid as each other -- which is why I left the warning to be the `<description>` in the metainfo file; it can be changed as appropriate.

Reply:
Hmm, it puts an implied requirement on the uploader of the alternate firmware to explain the implications for warranty or functionality in the description.  That seems a bit unfair to the original firmware creator.

Reply:
> it puts an implied requirement on the uploader of the alternate firmware to explain the implications for warranty or functionality in the description

Reply:
How about if you compare the vendor of the two different release branches and show such a boilerplate warning only if it's different?

Reply:
Plus, as we're only allowing specific OEMs on the LVFS access to the branch setting feature it's something we can pre-agree before allowing the user to upload the firmware.

Reply:
> compare the vendor of the two different release branches

That's not a bad idea at all. I suck at the prose stuff, so what about:

    This firmware was not supplied by the original hardware vendor
    Your hardware may be damaged using this firmware, and installing this
    release may void any warranty with the vendor.

Reply:
Sounds good to me, but you can probably switch out `vendor` for `%s` and use the vendor's name since we have it.

Reply:
So how about:

    The firmware supplied by {Hughski Limited} is not from {Broadcom Inc}, the hardware vendor. Your hardware may be damaged using this firmware, and installing this release may void any warranty with {Broadcom Inc}.

Reply:
maybe `s/not from/not supplied by`

Reply:
LGTM

Reply:
Fixed, thanks.